units/src/amount/unsigned.rs

```diff
@@ -227,12 +247,19 @@ impl Amount {
     ///
     /// Returns [`None`] if overflow occurred.
     #[cfg(feature = "alloc")]
-    pub fn checked_div_by_weight_ceil(self, rhs: Weight) -> Option<FeeRate> {
-        let sats = self.0.checked_mul(1_000)?;
+    pub const fn checked_div_by_weight_ceil(self, rhs: Weight) -> Option<FeeRate> {
         let wu = rhs.to_wu();
-
-        let fee_rate = sats.checked_add(wu.checked_sub(1)?)?.checked_div(wu)?;
-        Some(FeeRate::from_sat_per_kwu(fee_rate))
+        // No `?` operator in const context.
+        if let Some(sats) = self.0.checked_mul(1_000) {
+            if let Some(wu_minus_one) = wu.checked_sub(1) {
+                if let Some(sats_plus_wu_minus_one) = sats.checked_add(wu_minus_one) {
+                    if let Some(fee_rate) = sats_plus_wu_minus_one.checked_div(wu) {
+                        return Some(FeeRate::from_sat_per_kwu(fee_rate));
+                    }
+                }
+            }
+        }
+        None
```

Comment:
OMG that is horrible, I don't see a better way to do it though.

Reply:
Yeah, the only other ways I could think of were longer and not really any easier to follow.

Reply:
I hate to see what the corresponding match statement would look like.

Reply:
https://programmerhumor.io/wp-content/uploads/2022/05/programmerhumor-io-databases-memes-backend-memes-298f6b8ca108082.jpg